site/gatsby-site/src/pages/apps/classifications.js

```diff
@@ -311,6 +312,185 @@ const getClassificationsArray = (classifications, taxonomy) => {
   return array;
 };
 
+const formatDateField = (s) => {
+  const dateObj = new Date(s.props.cell.value);
+
+  if (dateObj instanceof Date && !isNaN(dateObj)) {
+    return <>{format(new Date(dateObj), 'yyyy-MM-dd')}</>;
+  } else {
+    return <>{s.props.cell.value}</>;
+  }
+};
+
+function EditForm({
+  allTaxonomies,
+  allClassifications,
+  row,
+  editFormRef,
+  onSubmit,
+  currentTaxonomy,
+}) {
+  const taxonomyFormObj = {
+    classificationsArray: [],
+    namespace: '',
+    taxonomyFields: [],
+  };
+
+  const taxaData = allTaxonomies.filter((taxa) => taxa.namespace === currentTaxonomy)[0];
+
+  taxonomyFormObj.namespace = taxaData.namespace;
+  taxonomyFormObj.taxonomyFields = taxaData.field_list.map((f) => {
+    return {
+      display_type: f.display_type,
+      long_name: f.long_name,
+      public: f.public,
+      short_description: f.short_description,
+      short_name: f.short_name,
+      weight: f.weight,
+    };
+  });
+
+  const classificationObj = allClassifications.filter(
+    (report) => report.incident_id === row.values.IncidentId
+  );
+
+  taxonomyFormObj.classificationsArray = getClassificationsArray(
+    classificationObj.length > 0 ? classificationObj[0].classifications : null,
+    taxaData
+  );
+
+  if (classificationObj.length === 1) {
+    taxonomyFormObj.notes = classificationObj[0].notes;
+  }
+
+  return (
+    <div className="bootstrap">
+      <TaxonomyForm
+        ref={editFormRef}
+        namespace={taxaData.namespace}
+        incidentId={row.values.IncidentId}
+        onSubmit={onSubmit}
+      />
+    </div>
+  );
+}
+
+function Row({
```

Comment:
This file is very long. Can we move some of the components into their own files?

Reply:
Opened this one https://github.com/responsible-ai-collaborative/aiid/issues/1563